lib/src/data/article/ny_times_cached_rest_repository.dart

```diff
@@ -0,0 +1,92 @@
+import 'package:allthenews/src/data/persistence/cache/cache_policy.dart';
+import 'package:allthenews/src/data/persistence/database/app_database.dart';
+import 'package:allthenews/src/data/persistence/database/article_dao.dart';
+import 'package:allthenews/src/data/persistence/database/article_dto_type_converter.dart';
+import 'package:allthenews/src/domain/model/article.dart';
+import 'package:allthenews/src/domain/nytimes/ny_times_repository.dart';
+import 'package:allthenews/src/domain/settings/popular_news_criterion.dart';
+import 'package:allthenews/src/domain/settings/settings_repository.dart';
+import 'package:moor/moor.dart';
+
+class NyTimesCachedRestRepository extends NYTimesRepository {
+  final NYTimesRepository _nyTimesRepository;
+  final ArticleDao _articleDao;
+  final CachePolicy<Article> _articleCachePolicy;
+  final SettingsRepository _settingsRepository;
+
+  NyTimesCachedRestRepository(
+    this._nyTimesRepository,
+    this._articleDao,
+    this._articleCachePolicy,
+    this._settingsRepository,
+  );
+
+  @override
+  Future<List<Article>> getMostPopularArticles() async {
+    final PopularNewsCriterion popularNewsCriterion = await _settingsRepository.getPopularNewsCriterion();
+    final Article latestCachedArticle = (await _articleDao.getLatestMostPopularArticle(popularNewsCriterion))?.toDomain();
```

Comment:
to może być wykonane równolegle

Reply:
chyba nie rozumiem, `_articleDao.getLatestMostPopularArticle` potrzebuje `popularNewsCriterion` :thinking:

Reply:
rzeczywiście, nie zauważyłem tego, w takim razie git